nix/haskell-dependencies.nix

```
@@ -0,0 +1,43 @@
+# Extract our the Haskell dependencies of sharkmachine
+# from the set of all haskell packages in nixpkgs,
+# and add the external packages that are not included in nixpkgs.
+# Whenever a new dependency is introduced, it has to be  added to the list below.
+#
+# `haskellPackages` is an attribute set containing all available Haskell packages
+# in the nixpkgs instantiation.
+haskellPackages:
+  with haskellPackages; [
+        aeson
+        aeson-pretty
+        blaze-html
+        blaze-markup
+        bytestring
+        containers
+        cryptonite
+        directory
+        extra
+        file-embed
+        filepath
+        free
+        github
+        hspec
+        hspec-core
+        http-client
+        http-conduit
+        http-types
+        memory
+        monad-logger
+        optparse-applicative
+        process
+        process-extras
+        scotty
+        stm
+        text
+        text-format
+        time
+        uuid
+        vector
+        wai
+        warp
+        warp-tls
+  ]
```

The indentation is a bit odd, how about this?

```suggestion
haskellPackages: with haskellPackages; [
  aeson
  aeson-pretty
  blaze-html
  blaze-markup
  bytestring
  containers
  cryptonite
  directory
  extra
  file-embed
  filepath
  free
  github
  hspec
  hspec-core
  http-client
  http-conduit
  http-types
  memory
  monad-logger
  optparse-applicative
  process
  process-extras
  scotty
  stm
  text
  text-format
  time
  uuid
  vector
  wai
  warp
  warp-tls
]
```

---

Copy&paste...